src/main/java/io/microprofile/config/ConfigProvider.java

```diff
@@ -13,28 +13,163 @@
+ * pick up the 'Configuration' for the Thread Context ClassLoader
+ * (See {@link  Thread#getContextClassLoader()}). </p>
+ *
+ * <p>A 'Configuration' consists of the information collected from the registered
+ * {@link ConfigSource}s. These {@link ConfigSource}s
+ * get sorted according to their <em>ordinal</em> defined via {@link ConfigSource#getOrdinal()}.
+ * That way it is possible to overwrite configuration with lower importance from outside.</p>
+ *
+ * <p>It is also possible to register custom {@link ConfigSource}s to
+ * flexibly extend the configuration mechanism. An example would be to pick up configuration values
+ * from a database table./p>
+ *
+ * <p>Example usage:
+ *
+ * <pre>
+ *     String restUrl = ConfigProvider.getConfig().getValue("myproject.some.remote.service.url");
+ *     Integer port = ConfigProvider.getConfig().getValue("myproject.some.remote.service.port", Integer.class);
+ * </pre>
+ *
+ * </p>
  *
+ * @author <a href="mailto:struberg@apache.org">Mark Struberg</a>
+ * @author <a href="mailto:rmannibucau@apache.org">Romain Manni-Bucau</a>
+ * @author <a href="mailto:emijiang@uk.ibm.com">Emily Jiang</a>
  */
-public interface ConfigProvider {
-
-	/**
-	 * Create a builder so that custom config resources can be added.
-	 * @return an instance {@link ConfigBuilder}
-	 */
-	ConfigBuilder builder();
-	/**
-	 * Get the {@link Config} after aggregating on multiple {@link ConfigSource}s
-	 * @return the {@link Config} containing all config properties
-	 */
-	Config getConfig();
+public class ConfigProvider {
+
+    private static final ConfigProviderResolver instance = ConfigProviderResolver.instance();
+
+    /**
+     * Provide a {@link Config} based on all {@link ConfigSource}s
+     * of the current Thread Context ClassLoader (TCCL)
+     *
+     * <p>There is exactly a single Config instance per ClassLoader</p>
+     */
+    public static Config getDefaultConfig() {
+        return instance.getDefaultConfig();
+    }
+
+    /**
+     * Provide a {@link Config} based on all {@link ConfigSource}s
+     * of the specified Thread Context ClassLoader (TCCL)
+     *
+     * <p>There is exactly a single Config instance per ClassLoader</p>
+     */
+    public static Config getConfig(ClassLoader cl) {
+        return instance.getConfig(cl);
+    }
+
+
+    /**
+     * Create a fresh {@link ConfigBuilder} instance.
+     * This ConfigBuilder will initially contain no
+     * {@link ConfigSource} nor any {@link Converters}.
+     * Those have to be added manually.
+     *
+     * The ConfigProvider will not manage the Config instance internally
+     */
+    public static ConfigBuilder emptyBuilder() {
+        return instance.emptyBuilder();
+    }
+    /**
+     * Create a {@link ConfigBuilder} instance.
+     * This builder contains the default {@link ConfigSource} and the {@link Converters}.
+     * @return the ConfigBuilder instance
+     */
+    public static ConfigBuilder builder() {    	
+    	return instance.builder();
```

Comment:
adding ConfigSources to what? The Config should be immutable. There should be no addConfigSource on the Config imo. Did I miss anything?